Laurence augmented security considerations on collections (#113)
ietf-rats-wg/draft-ietf-rats-msg-wrap: draft-ietf-rats-msg-wrap.md

This document introduces a format for holding multiple CMW items in a collection.
If the collection is not protected from tampering by external security measures (such as object security primitives) or internal mechanisms (such as intra-item binding), an attacker could easily manipulate the collection's contents.

> In particular, when a CMW is used to carry multiple Evidence messages for a composite device or layered attestation, there should be strong binding between the Evidence messages within the collection.

Reviewer: ```suggestion
In particular, when a CMW is used to carry multiple Evidence messages for a composite device or layered attestation, there SHOULD be strong binding between the Evidence messages within the collection.
```

Author: As discussed, with the new L590 these "should"s are clearly non-normative.

> In particular, when a CMW is used to carry multiple Evidence messages for a composite device or layered attestation, there should be strong binding between the Evidence messages within the collection.

Author: ```suggestion
It is the responsibility of the CMW collection producer to ensure that the contents of the collection are integrity protected.
In particular, when a CMW is used to carry multiple Evidence messages for a composite device or layered attestation, there should be strong binding between the Evidence messages within the collection.
```

This is to prevent attacks where Evidence from a subverted part of the device is replaced by Evidence from a separate non-subverted device.

> The binding of Evidence messages should be some form of attestation.

Reviewer: ```suggestion
The binding of Evidence messages SHOULD be some form of attestation.
```

Author: DITTO :-D

Reviewer: Ditto `DITTO` :o)

Author: ditto ditto ditto

This document introduces a format for holding multiple CMW items in a collection.
If the collection is not protected from tampering by external security measures (such as object security primitives) or internal mechanisms (such as intra-item binding), an attacker could easily manipulate the collection's contents.

> It is the responsibility of the CMW collection producer to ensure that the contents of the collection are integrity-protected.

Reviewer 2: ```suggestion
It is the responsibility of the Attester, which creates the CMW collection, to ensure that the contents of the collection are integrity-protected.
```

> It is the responsibility of the Attester who creates the CMW collection to ensure that the contents of the collection are integrity-protected.

Reviewer 2: ```suggestion
It is the responsibility of the Attester who creates the CMW collection to ensure that the contents of the collection are integrity-protected. The designer of the attestation technology is typically in charge of ensuring that the security properties are met, not the user of the conceptual message wrapper.
`
```

> It is the responsibility of the Attester, which creates the CMW collection, to ensure that the contents of the collection are integrity-protected.

Author: ```suggestion
It is the responsibility of the Attester who creates the CMW collection to ensure that the contents of the collection are integrity-protected.
```

> In particular, when a CMW is used to carry multiple Evidence messages for a composite device or layered attestation, there should be strong binding between the Evidence messages within the collection.

Reviewer 2: ```suggestion
In particular, when a CMW is used to carry multiple Evidence messages for a composite device or layered attestation, there should be strong binding between the Evidence messages within the collection. The designer of the attestation technology is typically in charge of ensuring that the security properties are met - not the user of the conceptual message wrapper.
```

Author: This one seems to break the natural flow between L595 ("In particular, ...") and L596 ("This is to prevent ...").

Reviewer 2: Fixed. Moved the sentence

> This is to prevent attacks where Evidence from a subverted part of the device is replaced by Evidence from a separate non-subverted device.

Reviewer 2: ```suggestion
This binding is to prevent attacks where Evidence from a subverted part of the device is replaced by Evidence from a separate non-subverted device.
```

The binding of Evidence messages should be some form of attestation.

> For example, key material used to sign/bind an entire CMW collection should be attestation key material handled as described in {{Section 12.1 of -rats-arch}}.

Reviewer 2: ```suggestion
For example, key material used to sign/bind an entire CMW collection chould be attestation key, as discussed in {{Section 12.1 of -rats-arch}}.
```

Author: I think the original is more accurate: S12.1 of the architecture discusses the requirements for attestation key material handling.
